libs/components/tiles/src/lib/modules/tiles/tile-dashboard/tile-dashboard.service.ts

```diff
@@ -89,38 +93,43 @@ export class SkyTileDashboardService {
     singleColumn?: SkyTileDashboardColumnComponent,
     settingsKey?: string,
   ): void {
-    if (settingsKey) {
-      // Clone this so changes to the config object outside of this class don't modify
-      // the config used inside and vice versa.
-      this.#config = Object.assign({}, config) as SkyTileDashboardConfig;
+    // Clone this so changes to the config object outside of this class don't modify
+    // the config used inside and vice versa.
+    this.#config = Object.assign({}, config) as SkyTileDashboardConfig;
+
+    // Verify that all tiles are accounted for in the layout.
+    const tileIds = [
+      ...new Set([
+        ...this.#config.layout.multiColumn.flatMap((column) =>
+          column.tiles.map(({ id }) => id),
+        ),
+        ...this.#config.layout.singleColumn.tiles.map(({ id }) => id),
+      ]),
+    ];
+    this.#checkForNewTiles(this.#config, tileIds);
```

Comment:
Handles the case where a tile is in the `tiles` config but omitted from one or both layouts, as was the case in the genesis of this issue.